armi/reactor/blocks.py

```diff
@@ -2344,36 +2318,40 @@ def autoCreateSpatialGrids(self):
                 )
             )
 
-        ringNumber = hexagon.numRingsToHoldNumCells(self.getNumPins())
+        nPins = self.getNumPins()
+        nRings = hexagon.numRingsToHoldNumCells(nPins)
         # For the below to work, there must not be multiple wire or multiple clad types.
         # note that it's the pointed end of the cell hexes that are up (but the
         # macro shape of the pins forms a hex with a flat top fitting in the assembly)
-        grid = grids.HexGrid.fromPitch(
-            self.getPinPitch(cold=True), numRings=0, cornersUp=True
-        )
-        spatialLocators = grids.MultiIndexLocation(grid=self.spatialGrid)
-        numLocations = 0
-        for ring in range(ringNumber):
-            numLocations = numLocations + hexagon.numPositionsInRing(ring + 1)
-        if numLocations != self.getNumPins():
+        if self.spatialGrid is not None:
+            grid = self.spatialGrid
+        else:
+            grid = grids.HexGrid.fromPitch(
```

Comment:
Do you want to tie this to the block by using the `armiObject` attribute? Just curious.

Reply:
That's a good idea